### DISCUSSION TITLE
Nested route on the same page

### DISCUSSION THREAD

**Discussion by t-zander** [vercel/next.js#50612] [App Router]
Let's say there is a page called `products`. When user goes to this page, it shows a list of products.
The structure would be smth like
```
app/
-- products/
---- pages.tsx
```

In UI it will look like this
![image](https://github.com/vercel/next.js/assets/34350867/8ed0b404-d71a-4e9e-acc5-7e228f4db42e)

Then, when user clicks on a specific product it should show that product details on the same page on the right side and url should change to `products/:id`
In UI it will look like this:
![image](https://github.com/vercel/next.js/assets/34350867/7073c052-ed54-47c3-9c8c-642bc34a706e)

In react world you would just have one top lvl `<Outlet />` to render `/products` and then you'd have a nested route and nested `<Outlet />` to render product information. Like this:
```
const routes = [
  {
    path: '/products'
    element: <ProductsList />
    children: [
      {
        path: '/:id',
        element: <ProductDetails />
      }
    ]
  }
]

// ProductsList component

return (
  ... showing products list
  <Outlet /> <--- to show product details on the same page
)
```

Is it possible to do it in next.js world? If I created another page like
```
app/
-- products/
---- page.tsx
---- [id]
------ page.tsx
```

It will be a completely separate page.
So I am unsure how to achieve this. It seems to be a common pattern in react, so hope it can be achieved in next.js 🙏  Thanks in advance for any suggestions.

**Comment by naseef0** [comment]
Have you tried `layout.js` here in products/layout.js?

**Reply by t-zander** [reply]
If I add a layout.js in `products/` I believe it will still only have access to `products/page.js`. 
So it won't let me render nested route under the same page.

**Reply by fernandoem88** [reply]
did you try /...products/page.tsx?

the in your page.tsx you can check the params
